### PR TITLE
Dryrun for duplicates

### PR DESCRIPTION
**Pull request by blob79**
```
czkawka_cli dup --dryrun -d $(pwd)/folder -m1 -D aen
Found 2 duplicated files in 1 groups with same content which took 2 B:
Size - 2 B (2) - 2 files
/home/thomas/Development/czkawka/folder/a
/home/thomas/Development/czkawka/folder/b
----

-------------------------------WARNINGS--------------------------------
Delete /home/thomas/Development/czkawka/folder/b
---------------------------END OF WARNINGS-----------------------------
```

### CODE REVIEW

File: czkawka_core/src/duplicate.rs [hunk 1]
```diff
@@ -1314,16 +1347,23 @@ fn delete_files(vector: &[FileEntry], delete_method: &DeleteMethod, warnings: &m
...
-                        warnings.push(format!("Failed to link {} -> {} ({})", file.path.display(), src.display(), e));
-                    } else {
-                        removed_files += 1;
-                        gained_space += file.size;
-                    }
+                if q_index == index {
+                    continue;
+                }
+                let r = if dryrun {
+                    warnings.push(format!("Replace file {} with hard link to {}", file.path.display(), src.display()));
```
Comment (qarmin) [inline czkawka_core/src/duplicate.rs]: `warnings` should be changed to `messages`.
Errors means non recoverable errors like e.g. not providing valid directories
Warnings informs about e.g. failing in remove, link, add some objects
Messages just are informational

**Review by qarmin — CHANGES_REQUESTED** [review]

**Comment by qarmin** [comment]
Also needs rebase after https://github.com/qarmin/czkawka/pull/278

File: czkawka_core/src/duplicate.rs (hunk 1 above)
Comment (blob79) [reply czkawka_core/src/duplicate.rs]: Makes sense.

**Review by qarmin — APPROVED** [review]

**Comment by qarmin** [comment]
Thanks!

**Pull request merged**